common/modsecurity.conf

```diff
@@ -0,0 +1,57 @@
+# Recommended configuration https://raw.githubusercontent.com/SpiderLabs/ModSecurity/v3/master/modsecurity.conf-recommended 
+
+SecRuleEngine ${MODSEC_RULE_ENGINE}
+SecRequestBodyAccess ${MODSEC_REQ_BODY_ACCESS}
+
+SecRequestBodyLimit ${MODSEC_REQ_BODY_LIMIT}
+SecRequestBodyNoFilesLimit ${MODSEC_REQ_BODY_NOFILES_LIMIT}
+SecRequestBodyInMemoryLimit 131072
+SecRequestBodyLimitAction Reject
+
+SecRule REQUEST_HEADERS:Content-Type "(?:application(?:/soap\+|/)|text/)xml" \
+     "id:'200000',phase:1,t:none,t:lowercase,pass,nolog,ctl:requestBodyProcessor=XML"
+SecRule REQUEST_HEADERS:Content-Type "application/json" \
+     "id:'200001',phase:1,t:none,t:lowercase,pass,nolog,ctl:requestBodyProcessor=JSON"
+SecRule REQBODY_ERROR "!@eq 0" \
+     "id:'200002', phase:2,t:none,log,deny,status:400,msg:'Failed to parse request body.',logdata:'%{reqbody_error_msg}',severity:2"
+SecRule MULTIPART_UNMATCHED_BOUNDARY "!@eq 0" \
+     "id:'200004',phase:2,t:none,log,deny,msg:'Multipart parser detected a possible unmatched boundary.'"
+SecRule TX:/^MSC_/ "!@streq 0" \
+     "id:'200005',phase:2,t:none,deny,msg:'ModSecurity internal error flagged: %{MATCHED_VAR_NAME}'"
+SecRule MULTIPART_STRICT_ERROR "!@eq 0" \
+     "id:'200003',phase:2,t:none,log,deny,status:400, \
+     msg:'Multipart request body failed strict validation: \
+     PE %{REQBODY_PROCESSOR_ERROR}, \
+     BQ %{MULTIPART_BOUNDARY_QUOTED}, \
+     BW %{MULTIPART_BOUNDARY_WHITESPACE}, \
+     DB %{MULTIPART_DATA_BEFORE}, \
+     DA %{MULTIPART_DATA_AFTER}, \
+     HF %{MULTIPART_HEADER_FOLDING}, \
+     LF %{MULTIPART_LF_LINE}, \
+     SM %{MULTIPART_MISSING_SEMICOLON}, \
+     IQ %{MULTIPART_INVALID_QUOTING}, \
+     IP %{MULTIPART_INVALID_PART}, \
+     IH %{MULTIPART_INVALID_HEADER_FOLDING}, \
+     FL %{MULTIPART_FILE_LIMIT_EXCEEDED}'"
+
+SecPcreMatchLimit ${MODSEC_PCRE_MATCH_LIMIT}
+SecPcreMatchLimitRecursion ${MODSEC_PCRE_MATCH_LIMIT_RECURSION}
+
+SecResponseBodyAccess ${MODSEC_RESP_BODY_ACCESS}
+SecResponseBodyMimeType text/plain text/html text/xml
+SecResponseBodyLimit ${MODSEC_RESP_BODY_LIMIT}
+SecResponseBodyLimitAction ProcessPartial
+
+SecTmpDir /tmp/
+SecDataDir /tmp/
+
+SecAuditEngine RelevantOnly
+SecAuditLogRelevantStatus "^(?:5|4(?!04))"
+SecAuditLogParts ABIJDEFHZ
+SecAuditLogType Serial
+SecAuditLog /var/log/modsec_audit.log
+
+SecArgumentSeparator &
+SecCookieFormat 0
+SecUnicodeMapFile unicode.mapping 20127
+SecStatusEngine On

```

Comment:
I's prefer this file to live in `src/<target-directory>` (i.e. `src/etc/modsecurity.d/`).

Mirroring the directory structure has two advantages:

* It is self-explanatory (aka _"Where does this file live in the target image?"_)
* Only one `COPY` command is needed in the `Dockerfile` to integrate all configuration files in the right place. (No updating needed in the future.)